crates/ruff_python_formatter/src/expression/expr_yield.rs

```diff
@@ -36,10 +36,10 @@ impl NeedsParentheses for ExprYield {
         parent: AnyNodeRef,
         _context: &PyFormatContext,
     ) -> OptionalParentheses {
-        if parent.is_stmt_return() || parent.is_expr_await() {
-            OptionalParentheses::Always
-        } else {
+        if parent.is_stmt_assign() {
```

Comment:
This should get a longer explanation about how only assignment right hand side and yield statements don't need parentheses and the special casing in the grammar, but we only need to handle assignment RHS here because StmtExpr doesn't add parentheses anyway (i just had to check that last part myself)

Reply:
Good point, added a comment describing the situation for both `expr_yield` and `expr_yield_from`. Thanks for the explanation in https://github.com/astral-sh/ruff/pull/5921#issuecomment-1645352816 as well, helps me get a clearer image of how the formatter is structured and what to watch out for.